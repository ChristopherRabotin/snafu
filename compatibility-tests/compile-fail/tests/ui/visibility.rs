// There are also happy-path tests

mod outer {
    pub mod inner {
        use snafu::prelude::*;

        #[derive(Debug, Snafu)]
        #[snafu(visibility(pub(crate)))]
        pub(crate) enum Error {
            PubCrate,
            #[snafu(visibility(pub(in crate::outer)))]
            PubInPath,
            #[snafu(visibility)]
            Private,
        }
    }

    fn private_is_applied() {
        let _ = self::inner::PrivateSnafu.build();
    }
}

fn pub_in_path_is_applied() {
    let _ = self::outer::inner::PubInPathSnafu.build();
}

fn private_is_applied() {
    let _ = self::outer::inner::PrivateSnafu.build();
}

fn main() {}
