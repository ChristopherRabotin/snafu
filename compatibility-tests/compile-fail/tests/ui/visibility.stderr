error[E0603]: unit struct `Private` is private
  --> $DIR/visibility.rs:19:30
   |
19 |         let _ = self::inner::Private.build();
   |                              ^^^^^^^ private unit struct
   |
note: the unit struct `Private` is defined here
  --> $DIR/visibility.rs:7:25
   |
7  |         #[derive(Debug, Snafu)]
   |                         ^^^^^
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: unit struct `PubInPath` is private
  --> $DIR/visibility.rs:24:33
   |
24 |     let _ = self::outer::inner::PubInPath.build();
   |                                 ^^^^^^^^^ private unit struct
   |
note: the unit struct `PubInPath` is defined here
  --> $DIR/visibility.rs:7:25
   |
7  |         #[derive(Debug, Snafu)]
   |                         ^^^^^
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0603]: unit struct `Private` is private
  --> $DIR/visibility.rs:28:33
   |
28 |     let _ = self::outer::inner::Private.build();
   |                                 ^^^^^^^ private unit struct
   |
note: the unit struct `Private` is defined here
  --> $DIR/visibility.rs:7:25
   |
7  |         #[derive(Debug, Snafu)]
   |                         ^^^^^
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0624]: associated function `build` is private
  --> $DIR/visibility.rs:19:38
   |
19 |         let _ = self::inner::Private.build();
   |                                      ^^^^^ private associated function

error[E0624]: associated function `build` is private
  --> $DIR/visibility.rs:24:43
   |
24 |     let _ = self::outer::inner::PubInPath.build();
   |                                           ^^^^^ private associated function

error[E0624]: associated function `build` is private
  --> $DIR/visibility.rs:28:41
   |
28 |     let _ = self::outer::inner::Private.build();
   |                                         ^^^^^ private associated function
